### DISCUSSION TITLE
Failing to get content of bucket with AWS SDK for Ruby

### DISCUSSION THREAD

**Discussion by DJViking** [minio/minio#14410] [Q&A] [answered]
Running MinIO with docker on OpenSUSE Leap 15.3
https://docs.min.io/docs/minio-docker-quickstart-guide.html

    docker run -p 9000:9000 -p 9001:9001 \
        --name minio1 \
        -v ~/minio/data:/data \
        -e "MINIO_ROOT_USER=username" \
        -e "MINIO_ROOT_PASSWORD=secret" \
        quay.io/minio/minio server /data --console-address ":9001"

Trying out the AWS Ruby API for interacting with my buckets.
https://docs.min.io/docs/how-to-use-aws-sdk-for-ruby-with-minio-server.html
https://docs.aws.amazon.com/sdk-for-ruby/v3/api/index.html

I can list the buckets just fine, but listing the content within a bucket fails.
Same goes for when I try the example in the "Use AWS SDK for Rubyt with MinIO", to upload a testfile to the bucket.

    2022-02-24 13:55:32 UTC testdata
    Traceback (most recent call last):
            63: from ./s3-wrapper.rb:14:in `<main>'
            62: from /usr/lib64/ruby/gems/2.5.0/gems/aws-sdk-s3-1.113.0/lib/aws-sdk-s3/client.rb:8345:in `list_objects'
            61: from /usr/lib64/ruby/gems/2.5.0/gems/aws-sdk-core-3.127.0/lib/seahorse/client/request.rb:72:in `send_request'
    /usr/lib64/ruby/2.5.0/net/http.rb:939:in `rescue in block in connect': Failed to open TCP connection to testdata.172.17.0.1:9000 (getaddrinfo: Name or service not known) (Seahorse::Client::NetworkingError)

My simple little ruby script:

    #!/usr/bin/env ruby

    require 'aws-sdk-s3'

    client = Aws::S3::Client.new(
      endpoint: 'http://172.17.0.1:9000',
      access_key_id: 'username',
      secret_access_key: 'secret'
    )

    response = client.list_buckets
    response.buckets.each do |bucket|
      puts "#{bucket.creation_date} #{bucket.name}"
    end

    client.list_objects(bucket:'testdata').each do |response|
      puts response.contents.map(&:key)
    end

It works fine listing the content of the bucket when I use the AWS CLI.
```
aws --endpoint-url http://172.17.0.1:9000 s3 ls
2022-02-24 14:55:32 testdata
aws --endpoint-url http://172.17.0.1:9000 s3 ls testdata

**Comment by harshavardhana** [comment, marked as answer]
Please set path style access in your ruby constructor @DJViking

**Reply by DJViking** [reply]
You are thinking about the s3_force_path_style to the Client class?
> :s3_force_path_style (Boolean) — default: false — When true, requests will always use path style. This can be useful for testing environments.

It does not work
```
/usr/lib64/ruby/gems/2.5.0/gems/aws-sdk-core-3.127.0/lib/seahorse/client/configuration.rb:168:in `rescue in block in apply_options': invalid configuration option `:s3_force_path_style' (ArgumentError)
```

**Reply by DJViking** [reply]
I was looking at the wrong documentation
https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Client.html

Worked when I set the `force_path_style`.
